xref-token/src/xref.rs

```diff
@@ -31,18 +31,23 @@ impl Contract {
     }
 
     /// return the amount of to be distribute reward this time
-    pub(crate) fn try_distribute_reward(&self, cur_timestamp: Timestamp) -> Balance {
-        assert!(cur_timestamp >= self.prev_distribution_time, "Err: cur time less than prev time");
-        let ideal_amount = self.reward_per_sec * (nano_to_sec(cur_timestamp) - nano_to_sec(self.prev_distribution_time)) as u128;
-        min(ideal_amount, self.undistribute_reward)
+    pub(crate) fn try_distribute_reward(&self, cur_timestamp_in_sec: u32) -> Balance {
+        if cur_timestamp_in_sec > self.reward_genesis_time_in_sec && cur_timestamp_in_sec > self.prev_distribution_time_in_sec {
+            let ideal_amount = self.reward_per_sec * (cur_timestamp_in_sec - self.prev_distribution_time_in_sec) as u128;
+            min(ideal_amount, self.undistribute_reward)
+        } else {
+            0
+        }
     }
 
     pub(crate) fn distribute_reward(&mut self) {
-        let cur_time = env::block_timestamp();
+        let cur_time = nano_to_sec(env::block_timestamp());
         let new_reward = self.try_distribute_reward(cur_time);
-        self.undistribute_reward -= new_reward;
-        self.locked_token_amount += new_reward;
-        self.prev_distribution_time = cur_time;
+        if new_reward > 0 {
```

Comment:
is this to prevent overflow? how could new_reward be negative?

Reply:
First, `new_reward` and even new `self.undistribute_reward` could NOT be negative, it is promised by `fn try_distribute_reward(...)`;
Second, the if here is to save some storage write action in case of new_reward is 0.